datafusion-cli/Cargo.toml

```diff
@@ -27,7 +27,7 @@ repository = "https://github.com/apache/arrow-datafusion"
 rust-version = "1.58"
 
 [dependencies]
-clap = "2.33"
+clap = "3.0"
```

Comment:
I can understand why I got the error
```
➜  datafusion-cli git:(update_dep) ✗ cargo build
   Compiling clap v3.0.10
   Compiling datafusion-cli v5.1.0 (/Users/bytedance/arrow-datafusion/datafusion-cli)
error[E0432]: unresolved import `clap::crate_version`
  --> datafusion-cli/src/main.rs:18:12
   |
18 | use clap::{crate_version, App, Arg};
   |            ^^^^^^^^^^^^^ no `crate_version` in the root

error: cannot determine resolution for the macro `crate_version`
  --> datafusion-cli/src/main.rs:36:18
   |
36 |         .version(crate_version!())
   |                  ^^^^^^^^^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

For more information about this error, try `rustc --explain E0432`.
error: could not compile `datafusion-cli` due to 2 previous errors
```
cc @alamb

Reply:
probably because they removed `crate_version` from the public api in 3.x

Reply:
```
#[cfg(feature = "cargo")]
#[macro_export]
macro_rules! crate_version {
    () => {
        env!("CARGO_PKG_VERSION")
    };
}
```
I know the reason, we should use the 'cargo' feature